inttest/common/footloosesuite.go

```diff
@@ -412,19 +429,80 @@ func (s *FootlooseSuite) InitController(idx int, k0sArgs ...string) error {
 		return err
 	}
 	defer ssh.Disconnect()
+
+	if err := s.launchDelegate.InitController(ssh, k0sArgs...); err != nil {
+		s.T().Logf("failed to start k0scontroller on %s: %v", controllerNode, err)
+		return err
+	}
+
+	return s.WaitForKubeAPI(controllerNode, getDataDirOpt(k0sArgs))
+}
+
+// initControllerStandalone initializes a controller in 'standalone' mode, meaning that
+// the k0s executable is launched directly (vs. started from a service)
+func (s *FootlooseSuite) initControllerStandalone(conn *SSHConnection, k0sArgs ...string) error {
 	umaskCmd := ""
 	if s.ControllerUmask != 0 {
 		umaskCmd = fmt.Sprintf("umask %d;", s.ControllerUmask)
 	}
+
 	// Allow any arch for etcd in smokes
-	startCmd := fmt.Sprintf("%s ETCD_UNSUPPORTED_ARCH=%s nohup %s controller --debug %s >/tmp/k0s-controller.log 2>&1 &", umaskCmd, runtime.GOARCH, s.K0sFullPath, strings.Join(k0sArgs, " "))
-	_, err = ssh.ExecWithOutput(startCmd)
+	cmd := fmt.Sprintf("%s ETCD_UNSUPPORTED_ARCH=%s nohup %s controller --debug %s >/tmp/k0s-controller.log 2>&1 &", umaskCmd, runtime.GOARCH, s.K0sFullPath, strings.Join(k0sArgs, " "))
+
+	if _, err := conn.ExecWithOutput(cmd); err != nil {
+		return fmt.Errorf("unable to execute '%s': %w", cmd, err)
+	}
+
+	return nil
+}
+
+// initControllerOpenRC initializes a controller in 'openrc' mode, meaning that
+// the k0s executable is launched as a service managed by openrc.
+func (s *FootlooseSuite) initControllerOpenRC(conn *SSHConnection, k0sArgs ...string) error {
```

Comment:
Always asking myself when seeing those test method signatures: Since we have access to the testing.T here, wouldn't it be better to not return an error, but use `s.Require().NoError(...)` directly.  Is there a situation in which a test shouldn't abort if this method fails?

Reply:
Its mainly about not having to do `s.Require().NoError(...)` through a number of layers of functions. These also aren't explicitly test functions, but supporting 'code' for setting up the test. I like to have the actual assertions done in the proper `Test*(...)` functions instead of the supporting code. Just a matter of preference.